src/emulation/mod.rs

```diff
@@ -94,8 +96,11 @@ impl SyscallMock {
     }
 
     fn push_error(&mut self, error: String) {
-        if self.errors.is_none() {
-            self.errors = Some(error);
+        match self.errors {
```

Comment:
From what I see `TestResult` can only store a single error. How come you use the plural `errors`? The name together with the fact that this method is called `push_error` confused me a bit when reading this method, as I was expecting this to add an error to some list of errors. However, if there is an error already, the new error is ignored.

Reply:
Good point. I've changed the names. Is this clearer now?

Reply:
Yeah, it reads better to me!